crates/router/src/core/payments/tokenization.rs

```diff
@@ -208,9 +208,15 @@ where
                     .await?
                 };
 
-                let pm_card_details = resp.card.as_ref().map(|card| {
-                    PaymentMethodsData::Card(CardDetailsPaymentMethod::from(card.clone()))
-                });
+                #[cfg(feature = "v2")]
+                let pm_card_details = match &resp.payment_method_data {
+                    Some(api::PaymentMethodResponseData::Card(card_data)) => Some(
```

Comment:
Everywhere we are doing match on `resp.payment_method_data` and some data out for card variant, instead can we do one match and possible to get all the values?

Reply:
This is being used in tokenization flow in payments, did not know what to do of payments with regard to v2 so made the corresponding type change based on version for now.